code/modules/mob/living/carbon/human/emote.dm

```diff
@@ -62,45 +62,43 @@
 			playsound(Y,'sound/effects/thunder.ogg', 90, 1)
 			spawn(10)
 				user.gib()
-		if(B.contents.len)
-			var/obj/item/O = pick(B.contents)
-			var/turf/location = get_turf(B)
+		var/obj/item/weapon/storage/internal/pocket/butt/theinv = B.inv
+		if(theinv.contents.len)
+			var/obj/item/O = pick(theinv.contents)
 			if(istype(O, /obj/item/weapon/lighter))
 				var/obj/item/weapon/lighter/G = O
-				if(G.lit && location)
-					new/obj/effect/hotspot(location)
-					playsound(user, 'sound/misc/fart.ogg', 50, 1, 5)
+				if(G.lit && user.loc)
+					new/obj/effect/hotspot(user.loc)
+					playsound(user.loc, 'sound/misc/fart.ogg', 50, 1, 5)

```

Comment:
Well you added on ".loc" to all of the users, I don't think this is necessary because playsound() finds the appropriate turf by itself